client/src/test/java/com/defold/extender/client/ExtenderClientTest.java

```diff
@@ -373,6 +377,50 @@ public CloseableHttpResponse execute(HttpUriRequest request) {
         }
     }
 
+    @Test(expected = ExtenderClientException.class)
+    public void testClientHandleHTTPError() throws ClientProtocolException, IOException, ExtenderClientException {
```

Comment:
Can't make the same simulation as it provided in initial issue. But at least now we have the test for HTTP error handling.